File: Ejercicio4/filtermanager.py
from filterlist import FilterList
from filter import Filter

class FilterManager:
    def __init__(self,target):
        self.filterChain = FilterList()
        self.filterChain.setTarget(target)

    def add_filter(self,filter):
        self.filterChain.addFilter(filter)

    def execute_on(self,credentials):
        self.filterChain.execute(credentials)

File: Practica2/ejercicio_conjunto_filtros/lib/filterManager.dart
import 'package:ejercicio_conjunto_filtros/FilterList.dart';
import 'package:ejercicio_conjunto_filtros/FormController.dart';
import 'package:ejercicio_conjunto_filtros/filters.dart';

class FilterManager {
  FilterList filterChain = FilterList();

  void addFilter(Filter filter) {
    filterChain.addFilter(filter);
    //TODO: target logic
  }

  void executeOn(FormController credentials) {
    filterChain.execute(credentials);
  }
}